### ISSUE TITLE
Manifold-json: java compiler complaining about interface beeing already defined

### ISSUE THREAD

**Issue by ltanguy** [manifold-systems/manifold#509]
**Describe the bug**
When using Manifold-json on a JSON Schema, Java throws an error at compile time about an interface being already defined. 

**To Reproduce**
Import the attached file (renamed with a .txt extension to make Github happy) in a project
[JiraIssuesResponse.json.txt](https://github.com/manifold-systems/manifold/files/12761850/JiraIssuesResponse.json.txt)

The Java compiler throws the following error:
```
java: interface JiraIssuesResponse.issues.issuesItem.fields is already defined in interface JiraIssuesResponse.issues.issuesItem
```

Already tried to clean the project, reload the maven project and rebuild the project without success.

**Expected behavior**
The code compiles.

**Screenshots**
If applicable, add screenshots to help explain your problem (drag/drop them here).

**Desktop (please complete the following information):**
- OS Type & Version: Windows 10 x64
- Java/JDK version: Temurin 17.0.8+1
- IDE version (IntelliJ IDEA or Android Studio): 2023.2.2
- Manifold version: 2023.1.27
- Manifold IntelliJ plugin version: 2023.2.3

**Additional context**

**Stack trace**
Please include a stack trace if applicable

**Comment by rsmckinney** [comment]
This is due to a limitation in Java where a nested class can't have the same name as any of its outer classes. The following is illegal.
```java
public class A {
    public class B {
        public class A {  // compile error: duplicate class
        }
    }
}
```
There is a `fields` interface nested inside another `fields` interface in the type projected for the JSON file.  Apologies for not accounting for this. It will be fixed in next week's build.

Thanks for reporting this!

**Comment by CC007** [comment]
I looked at the implementation and found that the solution might not fully fix the issue. See my comment here: https://github.com/manifold-systems/manifold/commit/32ba7fdd186195752926b16a1dbade909ed93f05#r128939227

**Comment by rsmckinney** [comment]
Fix available with release 2024.1.5
